Services/UI/OJS.Services.Ui.Models/Submissions/FullDetailsPublicSubmissionsServiceModel.cs

```diff
@@ -65,14 +63,41 @@ public void RegisterMappings(IProfileExpression configuration)
                 x => x.IsOfficial,
                 opt => opt.MapFrom(
                     y => y.Participant!.IsOfficial))
-            .ForMember(
-                d => d.MaxMemoryUsed,
-                opt => opt.MapFrom(s => s.TestRuns.Any() ? s.TestRuns.Max(testRun => testRun.MemoryUsed) : (long?)null))
-            .ForMember(
-                d => d.MaxTimeUsed,
-                opt => opt.MapFrom(s =>
-                    s.TestRuns.Any() ? s.TestRuns.Max(testRun => testRun.TimeUsed) : (int?)null))
+            .ForMember(x => x.MaxMemoryUsed, opt => opt.MapFrom(
+                s => GetMaxMemoryAndTimeUsed(s.TestRunsCache).MaxMemoryUsed))
+            .ForMember(x => x.MaxTimeUsed, opt => opt.MapFrom(
```

Comment:
Is the parsing translate to SQL and done in the query? I suggest using `.AfterMap()`, as you only need the TestRunsCache, which is already present in the destination.

Reply:
Done.